src/context/mapStateSlice/index.ts

```diff
@@ -75,12 +75,12 @@ export const mapStateSlice = createSlice({
       { payload }: PayloadAction<LayerType>,
     ) => ({
       ...rest,
-      layers: layers.filter(({ id, group }) =>
+      layers: layers.filter(({ id, group }) => {
```

Comment:
why this change?

Reply:
there was a bug for the previous approach and need to update the `removeLayer` function by adding some checker and return statement. seems i forgot to revert the return statement back. fixed it in the latest commit